ddtrace/contrib/django/patch.py

```diff
@@ -89,19 +89,18 @@ def cursor(django, pin, func, instance, args, kwargs):
 
 
 def instrument_dbs(django):
-    def set_connection(wrapped, instance, args, kwargs):
-        _, conn = args
+    def get_connection(wrapped, instance, args, kwargs):
```

Comment:
I wonder if patching on get is really enough. What if one gets a reference to a connection before patching, and that reference is then used to perform DB operations?

Reply:
The patching of the get method here would happen as the app populate method when the app populate method is called. IIUC this would all happen before a user would have been able to get a reference to a connection.

A potential problem with patching connections already created here would be to avoid triggering creating connections that haven't already been created:

https://github.com/django/django/blob/1710cdbe79c90665046034fe1700933d038d90ad/django/utils/connection.py#L56-L64

@P403n1x87 worth still trying to do this patching?

Reply:
would be the same issue with set no? if something gets added before patching?

Reply:
Sorry, didn't get around to answer this.

> would be the same issue with set no? if something gets added before patching?

Indeed. That's why I thought the solution was to patch all registered connections as soon as `django` is patched, by adding a loop over `connections.all()`, then patch all connections whenever they are set. I think @majorgreys looked into this approach but was not valid?

Reply:
The problem with looping over `connections.all()` is that it can end up creating a connection:

https://github.com/django/django/blob/1710cdbe79c90665046034fe1700933d038d90ad/django/utils/connection.py#L75-L76
https://github.com/django/django/blob/1710cdbe79c90665046034fe1700933d038d90ad/django/utils/connection.py#L56-L64
https://github.com/django/django/blob/1710cdbe79c90665046034fe1700933d038d90ad/django/db/utils.py#L200-L205

We'd have to access the internal connections collection. I'm interested in seeing what a test case for this would look like though since the db connections would have to be referenced before our patching code would have been applied.

Reply:
Ah I see. Indeed the best thing would be to loop over `_connections`. It's maybe unlikely to happen, but what if one decides to patch `django` on the basis of a value stored in a DB? You'd need to a reference first; a bit of a stretch maybe.